### PR TITLE
Specify that each is wrt enclosing array for its position

### LINKED ISSUE #2630: Specification of `each`
Is the following model valid?
```
model DimensionTest
  constant Real ref[2] = {1.9, 3.0};
  
  model A
    Real x[2](start = ref[1:2]);
  equation
    der(x) = -x;
  end A;
  
  A a[3](each x(start = ref[1:2]));  
end DimensionTest;
```
My understanding is that it is not. My reasoning is the following. `each` applies the modifier `ref[1:2]` to all elements of the enclosing array. In the context of `a.x.start` the dimension of the enclosing array are `[3, 2]` therefore the modifier is expected to be a scalar.

The same reasoning would apply to the following model:
```
model DimensionTest2
  type B = Real[2];

  B b[3](each start = {1, 2});
equation
  der(b) = -b;
end DimensionTest2;
```

### PR DESCRIPTION
**Pull request by HansOlsson**
And add example.

Closes #2630

### CODE REVIEW

**Review by henrikt-ma — COMMENTED** [review]
I didn't see a clear conclusion in #2630 that this is supposed to implement.  Still, the examples look useful and can probably help bringing #2630 closer to a conclusion.

**Comment by HansOlsson** [comment]
> I didn't see a clear conclusion in #2630 that this is supposed to implement. Still, the examples look useful and can probably help bringing #2630 closer to a conclusion.

I agree that there wasn't a clear conclusion yet in #2630, but one of the possibilities (that I believe is the right one) was to make it clarify it as suggested.

**Comment by HansOlsson** [comment]
@henrikt-ma I can move the controversial part to another PR can this one then be accepted?

**Comment by henrikt-ma** [comment]
> @henrikt-ma I can move the controversial part to another PR can this one then be accepted?

Sounds like a plan!

**Review by eshmoylova — APPROVED** [review]

File: chapters/inheritance.tex [hunk 1]
```diff
@@ -551,6 +552,16 @@ \subsection{Modifiers for Array Elements}\label{modifiers-for-array-elements}
 \end{lstlisting}
 This implies \lstinline!b[k].c[i].a[j] = j!, \lstinline!b[k].c[i].d = i!, and \lstinline!b[k].p = k!.  For \lstinline!c.a! the additional (outer) \lstinline!each! has no effect,
 but it is necessary for \lstinline!c.d!.
+
+Subscripts on types behave the same as array variables.
```
Comment (henrikt-ma) [inline chapters/inheritance.tex:556]: By _subscripts on types_, do you mean specifying array dimensions on the type rather than after the variable name?

**Review by henrikt-ma — CHANGES_REQUESTED** [review]
Requesting change, as I find the phrase _subscripts on types_ unclear.

File: chapters/inheritance.tex (hunk 1 above)
Comment (HansOlsson) [reply chapters/inheritance.tex]: Yes. Updated accordingly.

**Review by eshmoylova — APPROVED** [review]

File: chapters/inheritance.tex (hunk 1 above)
Comment (henrikt-ma) [reply chapters/inheritance.tex]: Nice! Resolving.

**Review by henrikt-ma — APPROVED** [review]

**Pull request merged**